cli/main.rs

```diff
@@ -545,3 +568,75 @@ fn init_logging(
     on_log_end: DrawThread::show,
   })
 }
+
+#[cfg(not(unix))]
+async fn wait_for_start() -> Result<
+  Option<(deno_runtime::Unconfigured, Vec<std::ffi::OsString>)>,
+  AnyError,
+> {
+  Ok(None)
+}
+
+#[cfg(unix)]
+async fn wait_for_start() -> Result<
+  Option<(deno_runtime::Unconfigured, Vec<std::ffi::OsString>)>,
+  AnyError,
+> {
+  use std::os::unix::ffi::OsStringExt;
+
+  use tokio::io::AsyncReadExt;
+
+  let Some(startup_snapshot) = deno_snapshots::CLI_SNAPSHOT else {
```

Comment:
Since these two check are sync maybe you could rewrite this method somehow to not insert an await point for every startup when most of the time it's not necessary and we're gonna run from snapshot